src/redux/logging/loggingMiddleware.ts

```diff
@@ -0,0 +1,109 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+import { Topic } from 'openspace-api-js';
+
+import { api } from '@/api/api';
+import { LogLevel } from '@/types/enums';
+import { showNotification } from '@/util/logging';
+
+import { onCloseConnection, onOpenConnection } from '../connection/connectionSlice';
+import { AppStartListening } from '../listenerMiddleware';
+
+let topic: Topic | null = null;
+
+/** This `LogLevel` must match OpenSpace/Ghoul LogLevel in
+ * https://github.com/OpenSpace/Ghoul/blob/f02810ad2f77166711f4503060e20745f0d808c6/include/ghoul/logging/loglevel.h#L41
+ */
```

Comment:
Maybe nitpicky, but comment style is important
```suggestion
/** 
 * This `LogLevel` must match OpenSpace/Ghoul LogLevel in
 * https://github.com/OpenSpace/Ghoul/blob/f02810ad2f77166711f4503060e20745f0d808c6/include/ghoul/logging/loglevel.h#L41
 */
```